pkg/file/redundancy/getter/strategies.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package getter
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+var (
+	StrategyTimeout = 500 * time.Millisecond // timeout for each strategy
+)
+
+type (
+	strategyKey       struct{}
+	modeKey           struct{}
+	fetcherTimeoutKey struct{}
+	Strategy          = int
+)
+
+const (
+	NONE Strategy = iota // no prefetching and no decoding
+	DATA                 // just retrieve data shards no decoding
+	PROX                 // proximity driven selective fetching
+	RACE                 // aggressive fetching racing all chunks
+	strategyCnt
+)
+
+// GetParamsFromContext extracts the strategy and strict mode from the context
+func GetParamsFromContext(ctx context.Context) (s Strategy, strict bool, fetcherTimeout time.Duration) {
+	s, _ = ctx.Value(strategyKey{}).(Strategy)
+	strict, _ = ctx.Value(modeKey{}).(bool)
+	fetcherTimeout, _ = ctx.Value(fetcherTimeoutKey{}).(time.Duration)
+	return s, strict, fetcherTimeout
+}
+
+// SetFetchTimeout sets the timeout for each fetch
+func SetFetchTimeout(ctx context.Context, timeout time.Duration) context.Context {
+	return context.WithValue(ctx, fetcherTimeoutKey{}, timeout)
+}
+
+// SetStrategy sets the strategy for the retrieval
+func SetStrategy(ctx context.Context, s Strategy) context.Context {
+	return context.WithValue(ctx, strategyKey{}, s)
+}
+
+// SetStrict sets the strict mode for the retrieval
+func SetStrict(ctx context.Context, strict bool) context.Context {
+	return context.WithValue(ctx, modeKey{}, strict)
+}
+
+func (g *decoder) prefetch(ctx context.Context, strategy int, strict bool, strategyTimeout, fetchTimeout time.Duration) {
+	if strict && strategy == NONE {
+		return
+	}
+	defer g.remove()
+	var cancels []func()
+	cancelAll := func() {
+		for _, cancel := range cancels {
+			cancel()
+		}
+	}
+	defer cancelAll()
+	run := func(s Strategy) error {
+		if s == PROX { // NOT IMPLEMENTED
```

Comment:
what happens if strategy is DATA? because the strategy is incremented below, then for cases where the strategy is DATA, no other strategy is tried because PROX is not implemented ?